File: Python/July_problems/problem1717_12th_july_2024.py
def maximum_gain(s, x, y):
    larger = max(x, y)
    gain = 0

    if larger == x:
        while 'ab' in s:
            s = s.replace('ab', '', 1)
            gain += x
        while 'ba' in s:
            s = s.replace('ba', '', 1)
            gain += y
    else:
        while 'ba' in s:
            s = s.replace('ba', '', 1)
            gain += y
        while 'ab' in s:
            s = s.replace('ab', '', 1)
            gain += x

    return gain

File: CPP/Year-2024/July_Problems/problem1717_12th_july_2024.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int maximumGain(string s, int x, int y)
{
    int larger = max(x, y);
    int gain = 0;
    size_t pos;

    if (larger == x)
    {
        while ((pos = s.find("ab")) != string::npos)
        {
            s.erase(pos, 2);
            gain += x;
        }
        while ((pos = s.find("ba")) != string::npos)
        {
            s.erase(pos, 2);
            gain += y;
        }
    }
    else
    {
        while ((pos = s.find("ba")) != string::npos)
        {
            s.erase(pos, 2);
            gain += y;
        }
        while ((pos = s.find("ab")) != string::npos)
        {
            s.erase(pos, 2);
            gain += x;
        }
    }
    return gain;
}

int main()
{
    cout << maximumGain("cdbcbbaaabab", 4, 5);
    return 0;
}